smart_open/tests/test_smart_open.py

```diff
@@ -882,17 +913,22 @@ def test_open_gz(self):
         assert m.hexdigest() == '18473e60f8c7c98d29d65bf805736a0d', \
             'Failed to read gzip'
 
-    def test_write_read_gz(self):
-        """Can write and read gzip?"""
-        with tempfile.NamedTemporaryFile('wb', suffix='.gz', delete=False) as infile:
+    def _test_write_read(self, suffix):
```

Comment:
This function feels unnecessary. All it does is declare test_file_name and pass it into the write_read_assertion function.

Why not declare test_file_name in the write_read_assertion function directly? That would make the tests easier to read (need to step into only one extra function call instead of two).

Reply:
It used to be much worse before - two archive test function copy-pasted. NP, I will change it.